### PR TITLE
Add DDNet

### PR DESCRIPTION
**Pull request by ImKringle**
Was mentioned in #2836 - now closed

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-03-31T18:27:28-04:00",
+    "name": "DDNet",
+    "author": "imkringle@proton.me",
+    "description": null,
+    "features": null,
+    "docker_images": {
+        "Source": "ghcr.io\/parkervcp\/games:source"
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:13]: This is not a source game so use the steamcmd image.

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Steam AppID",
+            "description": "",
+            "env_variable": "SRCDS_APPID",
+            "default_value": "412220",
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|string",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:48]: Append here the in: validation

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Steam User",
+            "description": "",
+            "env_variable": "STEAM_USER",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:54]: Login is needed?

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Max Players",
+            "description": "",
+            "env_variable": "MAX_PLAYERS",
+            "default_value": "64",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:64",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:118]: Use between here

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-03-31T18:27:28-04:00",
+    "name": "DDNet",
+    "author": "imkringle@proton.me",
+    "description": null,
+    "features": null,
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:11]: Add the steam disk feature here

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json (hunk 5 above)
Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:11]: and a description maybe?

Comment (ImKringle) [reply game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json]: Added Description, Feature

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json (hunk 4 above)
Comment (ImKringle) [reply game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json]: Added Between validation

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json (hunk 3 above)
Comment (ImKringle) [reply game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json:56]: Login is required, Free game and the server executable is packaged into it. SteamDB treats it as *Free on Demand* but you can't pull as anonymous; tried.

File: game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json (hunk 2 above)
Comment (ImKringle) [reply game_eggs/steamcmd_servers/ddnet/egg-d-d-net.json]: Added in: validation

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! Thanks

**Pull request merged**